Formula/gnunet.rb

```diff
@@ -0,0 +1,36 @@
+class Gnunet < Formula
+  desc "GNUnet"
+  homepage "https://gnunet.org/"
+  url "https://ftp.gnu.org/gnu/gnunet/gnunet-0.11.0.tar.gz"
+  sha256 "b7477a3c3b0d5e8a013685dc208cfb4ccee4145f8668faa8eb5b382af36c7e9a"
+
+  depends_on "pkg-config" => :build
+  depends_on "gettext"
+  depends_on "gnutls"
+  depends_on "jansson"
+  depends_on "libextractor"
+  depends_on "libgcrypt"
+  depends_on "libidn2"
+  depends_on "libmicrohttpd"
+  depends_on "libmpc"
+  depends_on "libunistring"
+  depends_on "unbound"
+
+  def install
+    args = %W[
+      --disable-documentation
+      --prefix=#{prefix}
+    ]
+
+    system "./configure", *args
+    system "make", "install"
+  end
+
+  def post_install
+    chmod "+x", prefix/"bin/gnunet-qr.py"
+  end
+
+  test do
+    system bin/"gnunet-config", "-s", "arm"
```

Comment:
We need a test that exercises the some of the functionality of the app. What does this test do?

Reply:
I changed it to starting/stopping the gnunet services now.

Reply:
As I suspected starting the service will cause the tests to hang and gnunet will not be stopped.
My original proposed test "system bin/"gnunet-config", "-s", "arm""
basically loads the main gnunet config and prints it. Is that enough for a test?

Reply:
The original test can work, if we can match something in its output.

Reply:
Okay. Well if the output is not empty and the return code == 0 then it is successful.
It is not clear to me what we should do here. e.g. gnutls.rb simply does this:
```system bin/"gnutls-cli", "--version"```
Now, we could do the same, but I don't see how this tests anything :)
Further, ```gnunet-arm --version``` has a return code of 1 for some reason... (it does work though)
So can you point me to either:
- how to match the output (ideally an example formula that does it) or
- how to "accept" a return code of 1 and we do it like gnutls

Reply:
A package that is comparable to GNUnet in functionality is IPFS (ipfs.rb). Which also does:
```system bin/"ipfs", "version"```
so does profanity.rb.
Now, I am also a fan of proper tests but it does not seem to be like this is a KO criteria for formulas?

Reply:
Use `assert_match` or `assert_equal`, they are used plenty in the other formulas

Reply:
I changed it to something similar to easyrpg-player.rb in 38535527b..9f86c722c

Reply:
Builds/checks seem to pass now. My last commit moves the python script to ```pkgshare``` and patches the ```gnunet-qr``` script which calls the python script in order to find it in the new location.